deps/llhttp/llhttp.gyp

```diff
@@ -0,0 +1,72 @@
+# This file is used with the GYP meta build system.
+# http://code.google.com/p/gyp/
+# To build try this:
+#   svn co http://gyp.googlecode.com/svn/trunk gyp
+#   ./gyp/gyp -f make --depth=`pwd` llhttp.gyp 
+#   ./out/Debug/test 
+{
+  'target_defaults': {
```

Comment:
1. If you only have 1 target, you don't need `target_defaults`, just put these in the target.
2. This is the tree's `common.gypi` - https://github.com/nodejs/node/blob/master/common.gypi#L99
   We use set `RuntimeLibrary` at  https://github.com/nodejs/node/blob/master/common.gypi#L139 and https://github.com/nodejs/node/blob/master/common.gypi#L214

So AFAICT all you need is in this whole file is:
```python
{
  'targets': [
    {
      'target_name': 'llhttp',
      'type': 'static_library',
      'include_dirs': [ 'include', ],
      'direct_dependent_settings': {
        'include_dirs': [ 'include', ],
      },
      'sources': [ 'src/llhttp.c', 'src/api.c', 'src/http.c', ],
    },
  ]
}
```

Reply:
Tbh, I just copy-pasted it from http_parser :joy: Given that `llhttp` can be built as a standalone library, wouldn't it make sense to still keep the compiler options in its gyp file?

Reply:
I worry about chasing down osbcure conflict with the gyp config in the future.
You could do:
`llhttp.gyp` - with the kruft and `includes: [ llhttp.gypi ]`
|
 -> `llhttp.gypi` - with just the target I wrote above.

Then in `node.gypi` refer only to `llhttp.gypi`